pkg/controller/common/reconciler.go

```diff
@@ -30,6 +33,36 @@ func (r *BaseIngressReconciler) ReconcileIngress(ctx context.Context, ci network
 
 	exposed := ci.GetSpec().Visibility == networkingv1alpha1.IngressVisibilityExternalIP
 	if exposed {
+		// get the namespace where operator has been installed
+		ns, err := k8sutil.GetOperatorNamespace()
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
+				// here knative-openshift-ingress may be running as locally so assuming default namespace as knative-serving
+				ns = "knative-serving"
+			} else {
+				return err
+			}
+		}
+		// update ServiceMeshMemberRole with the namespace info where knative routes created
+		smmr := &maistrav1.ServiceMeshMemberRoll{}
+		if err = r.Client.Get(ctx, types.NamespacedName{Name: "default", Namespace: ns + "-ingress"}, smmr); err != nil {
+			return err
+		}
+		smmr.Spec.Members = func(members []string, ns string) []string {
```

Comment:
Why wrap this in a closure?

Reply:
Added in closure because that particular code is to check members of smmr and append only when if `ns`(ci.GetNamespace()) doesn't exist and no where its reused.

Reply:
I guess that's fine but it makes the code a tad bit harder to read imo.

Reply:
Updated by removing closure function